retriever/endpointfinder/endpointfinder_test.go

```diff
@@ -33,27 +34,27 @@ func TestEndpointFetcher(t *testing.T) {
 		{
 			testName:             "unable to find miner on chain",
 			minerInfoNotFindable: true,
-			expectedErrString:    fmt.Errorf("looking up provider info: %w", errMinerNotFound).Error(),
+			expectedErrString:    fmt.Errorf("no http endpoints found for providers [%%s]: looking up provider info: %w", errMinerNotFound).Error(),
 		},
 		{
 			testName:          "unable to dial provider",
 			notDialable:       true,
-			expectedErrString: "querying transports: failed to dial: %s cannot connect to %s",
+			expectedErrString: "no http endpoints found for providers [%s]: querying transports: failed to dial: %s cannot connect to %s",
 		},
 		{
 			testName:                 "provider not listening on protocol",
 			notListeningOnTransports: true,
-			expectedErrString:        "querying transports: failed to negotiate protocol: protocols not supported: [/fil/retrieval/transports/1.0.0]",
+			expectedErrString:        "no http endpoints found for providers [%s]: querying transports: failed to negotiate protocol: protocols not supported: [/fil/retrieval/transports/1.0.0]",
 		},
 		{
 			testName:          "provider not serving http",
 			noHTTP:            true,
-			expectedErrString: endpointfinder.ErrHTTPNotSupported.Error(),
+			expectedErrString: fmt.Errorf("no http endpoints found for providers [%%s]: %w", endpointfinder.ErrHTTPNotSupported).Error(),
```

Comment:
This looks like it is expecting something that is incorrectly formatted. Shouldn't the format of the returned error message be fixed? This applies to most/all of the error strings here.

Reply:
There's two steps to these error messages because they're used in a `fmt.Sprintf()` below that inserts the provider ID and peer IDs of self+remote, hence the `%%` to skip the first level of fmt substitution here. There's also a trim for `%!(EXTRA` that splits off unidentified parameters, so the test ends up matching for these (if I print them out), which are what we expect:

```
no http endpoints found for providers [t010001]: looking up provider info: miner not found
no http endpoints found for providers [t010002]: querying transports: failed to dial: QmV8GjxetvVBEfuXEMvDCYQqPogmSiZNPGe676Gny7d8hF cannot connect to QmQEsGHyTfaFV2sBax8pooVhvuquhdzL1y23jaUqWhUvWQ
no http endpoints found for providers [t010003]: querying transports: failed to negotiate protocol: protocols not supported: [/fil/retrieval/transports/1.0.0]
no http endpoints found for providers [t010004]: provider does not support http
```